repositories/scenario_testrun.go

```diff
@@ -0,0 +1,81 @@
+package repositories
+
+import (
+	"context"
+	"time"
+
+	"github.com/Masterminds/squirrel"
+	"github.com/checkmarble/marble-backend/models"
+	"github.com/checkmarble/marble-backend/repositories/dbmodels"
+)
+
+type ScenarioTestRunRepository interface {
+	CreateTestRun(ctx context.Context, tx Transaction, testrunID string,
+		input models.ScenarioTestRunInput) error
+	GetByScenarioIterationID(ctx context.Context, exec Executor, scenarioID string) (models.ScenarioTestRun, error)
+	GetByID(ctx context.Context, exec Executor, testrunID string) (models.ScenarioTestRun, error)
+}
+
+type ScenarioTestRunRepositoryPostgresql struct{}
+
+func selectTestruns() squirrel.SelectBuilder {
+	return NewQueryBuilder().
+		Select(dbmodels.SelectScenarioTestRunColumns...).
+		From(dbmodels.TABLE_SCENARIO_TESTRUN)
+}
+
+func (repo *ScenarioTestRunRepositoryPostgresql) CreateTestRun(ctx context.Context,
+	tx Transaction, testrunID string, input models.ScenarioTestRunInput,
+) error {
+	if err := validateMarbleDbExecutor(tx); err != nil {
+		return err
+	}
+	err := ExecBuilder(
+		ctx,
+		tx,
+		NewQueryBuilder().Insert(dbmodels.TABLE_SCENARIO_TESTRUN).
+			Columns(
+				"id",
+				"scenario_iteration_id",
+				"expires_at",
+				"status",
+			).
+			Values(
+				testrunID,
+				input.ScenarioIterationId,
+				time.Now().Add(input.Period),
+				models.Up.String(),
+			),
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (repo *ScenarioTestRunRepositoryPostgresql) GetByScenarioIterationID(ctx context.Context,
+	exec Executor, scenarioIterationID string,
+) (models.ScenarioTestRun, error) {
+	if err := validateMarbleDbExecutor(exec); err != nil {
+		return models.ScenarioTestRun{}, err
+	}
+	return SqlToModel(
```

Comment:
Using the helper method SqlToModel will return an error wrapping `models.NotFoundError` if no row is scanned (see https://github.com/checkmarble/marble-backend/blob/main/repositories/sql_to_row.go#L85), seeing how you're planning to use it in the usecase you may want to use `SqlToOptionalModel` or `SqlToListOfModels` (and adapt the signature accordingly)